tests/Aspire.NATS.Net.Tests/AspireNatsClientExtensionsTests.cs

```diff
@@ -101,4 +114,104 @@ public void ConnectionNameWinsOverConfigSection(bool useKeyed)
         // the connection string from config should not be used since it was found in ConnectionStrings
         Assert.DoesNotContain("unused", connection.Opts.Url);
     }
+
+    [RequiresDockerTheory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task AddNatsClient_HealthCheckShouldBeRegisteredWhenEnabled(bool useKeyed)
+    {
+        var key = DefaultConnectionName;
+        var builder = CreateBuilder(DefaultConnectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(key, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+
+        var host = builder.Build();
+
+        var healthCheckService = host.Services.GetRequiredService<HealthCheckService>();
+
+        var healthCheckReport = await healthCheckService.CheckHealthAsync();
+
+        var healthCheckName = useKeyed ? $"NATS_{key}" : "NATS";
+
+        Assert.Contains(healthCheckReport.Entries, x => x.Key == healthCheckName);
+    }
+
+    [RequiresDockerTheory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void AddNatsClient_HealthCheckShouldNotBeRegisteredWhenDisabled(bool useKeyed)
+    {
+        var builder = CreateBuilder(DefaultConnectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = false;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = false;
+            });
+        }
+
+        var host = builder.Build();
+
+        var healthCheckService = host.Services.GetService<HealthCheckService>();
+
+        Assert.Null(healthCheckService);
+    }
+
+    [RequiresDockerFact]
+    public async Task NatsInstrumentationEndToEnd()
```

Comment:
IIUC, other tests here don't need `RequiresDocker*` attributes as they don't connect to the service.

Reply:
I believe they're correctly used where connection is required. could you be more specific if that's not the case?

Reply:
ah! got you. Thank you!